programs/drift-competitions/src/state/competitor.rs

```diff
@@ -23,4 +27,21 @@ impl Size for Competitor {
     const SIZE: usize = 104 + 8;
 }
 
-const_assert_eq!(Competitor::SIZE, std::mem::size_of::<Competitor>() + 8);
\ No newline at end of file
+const_assert_eq!(Competitor::SIZE, std::mem::size_of::<Competitor>() + 8);
+
+impl Competitor {
+    pub fn calculate_score(&self, user_stats: UserStats) -> DriftResult<u64> {
+        let current_snapshot_score = user_stats.fees.total_fee_paid;
+
+        let round_score = current_snapshot_score
+            .safe_sub(self.previous_snapshot_score)?
+            .safe_add(self.bonus_score)?;
+
+        Ok(round_score)
+    }
+
+    pub fn claim_entry(&mut self) -> DriftResult {
```

Comment:
you should only be able to do this once per round?

Reply:
either once per round or some logic to ensure you can stuff more than one in a single tx

Reply:
can we add an error to this function so we don't forget to add it?